pkg/generators/servers.go

```diff
@@ -330,19 +333,36 @@ func (g *ServersGenerator) generateServerAdapterSource(resource *concepts.Resour
 			{{ $requestBodyParameters := requestBodyParameters . }}
 			{{ $requestBodyLen := len $requestBodyParameters }}
 			{{ $responseParameters := responseParameters . }}
+			{{ $requestQueryParameters := requestQueryParameters . }}
 	
 			func (a *{{ $adapterName }}) read{{ $requestName }}(r *http.Request) (*{{ $requestName }}, error) {
+				var err error
 				result := new({{ $requestName }})
-				result.query = r.Form
+				result.query = r.URL.Query()
 				result.path = r.URL.Path
 
+				{{ if $requestQueryParameters }}
+					{{ range  $requestQueryParameters }}
```

Comment:
You may omit surrounding `if` — `range` handles empty slice, including nil slice correctly as no-op: https://play.golang.org/p/n2kpDRD9hf0

Reply:
Thanks!!